modules/lang-painless/src/main/java/org/elasticsearch/painless/PainlessScriptEngine.java

```diff
@@ -126,54 +118,24 @@ public String getType() {
     public <T> T compile(String scriptName, String scriptSource, ScriptContext<T> context, Map<String, String> params) {
         Compiler compiler = contextsToCompilers.get(context);
 
-        if (context.instanceClazz.equals(SearchScript.class)) {
-            Constructor<?> constructor = compile(compiler, scriptName, scriptSource, params);
```

Comment:
There are two versions of the compile method.  Now that the two special cases are gone for search/executable script contexts, would you please check to see if this version can be removed completely?  It's possible some tests still use it, though.